components/size_aware_image.jsx

```diff
@@ -110,18 +111,15 @@ export default class SizeAwareImage extends React.PureComponent {
         return width < MIN_IMAGE_SIZE || height < MIN_IMAGE_SIZE;
     }
 
-    handleLoad = (event) => {
+    handleLoad = () => {
         if (this.mounted) {
-            const image = event.target;
-            const isSmallImage = this.isSmallImage(image.naturalWidth, image.naturalHeight);
             this.setState({
                 loaded: true,
                 error: false,
-                isSmallImage,
-                imageWidth: image.naturalWidth,
+                imageWidth: this.props.dimensions.width,
```

Comment:
According to the prop types, `dimensions` isn't a required prop, so it might be null sometimes. I'm trying to remember if that comes from an old version of this from before the server sent information about the image to the clients. Are you sure this'll be defined, even if you call this with invalid or broken image links?

If that does end up being the case, we could probably make `dimensions` a required prop and remove `state.imageWidth` entirely. The reason that it was previously stored in state was just because the server didn't always send those dimensions previously

Reply:
Actually, I don't think `dimensions` is defined everywhere we use this component, like in the channel header or in the post preview. Basically, it happens anywhere that we use Markdown that isn't a post on the server. It also affects the "fake" posts we make while waiting for a server to fully create a post. We'll still need a fallback in these cases.

Reply:
Fix